src/js/components/state/visualizations/awardBreakdown/AwardBreakdownTreeMap.jsx

```diff
@@ -0,0 +1,265 @@
+/**
+ * AwardBreakdownTreeMap.jsx
+ * Created by Lizzie Salita 5/16/18
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { hierarchy, treemap, treemapBinary, treemapSlice } from 'd3-hierarchy';
+import { throttle, remove, orderBy, find } from 'lodash';
+import * as MoneyFormatter from 'helpers/moneyFormatter';
+import * as TreemapHelper from 'helpers/treemapHelper';
+import { awardTypeLabels } from 'dataMapping/state/awardTypes';
+import { labelColorFromBackground } from 'helpers/colorHelper';
+
+import AwardTypeCell from './AwardTypeCell';
+import AwardTypeTooltip from './AwardTypeTooltip';
+
+const propTypes = {
+    awardBreakdown: PropTypes.object,
+    totalAmount: PropTypes.number,
+    hasNegatives: PropTypes.bool
+};
+
+export default class AwardBreakdownTreeMap extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            windowWidth: 0,
+            visualizationWidth: 0,
+            visualizationHeight: 175,
+            finalNodes: [],
+            showOverlay: true,
+            hoveredAwardType: ''
+        };
+
+        this.handleWindowResize = throttle(this.handleWindowResize.bind(this), 50);
+        this.buildTree = this.buildTree.bind(this);
+        this.createTooltip = this.createTooltip.bind(this);
+        this.toggleTooltipIn = this.toggleTooltipIn.bind(this);
+        this.toggleTooltipOut = this.toggleTooltipOut.bind(this);
+    }
+
+    componentDidMount() {
+        this.handleWindowResize();
+        window.addEventListener('resize', this.handleWindowResize);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.awardBreakdown.children.length > 0) {
+            this.buildTree(nextProps);
+        }
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.handleWindowResize);
+    }
+
+    handleWindowResize() {
+        // determine if the width changed
+        const windowWidth = window.innerWidth;
+        if (this.state.windowWidth !== windowWidth) {
+            // width changed, update the visualization width
+            this.setState({
+                windowWidth,
+                visualizationWidth: this.sectionWrapper.offsetWidth
+            });
+            if (this.props.awardBreakdown.children.length > 0) {
+                this.buildTree(this.props);
+            }
+        }
+    }
+
+    buildTree(treeProps) {
```

Comment:
Can you update this to follow the `buildVirtualChart` approach we use in the spending explorer? https://github.com/fedspendingtransparency/usaspending-website/blob/dev/src/js/components/explorer/detail/visualization/treemap/ExplorerTreemap.jsx#L54

Basically, instead of generating SVG components and saving them in state, we create JS objects that have all the calculated values the SVG components will need and then put the actual SVG in either standalone components or in the render output. The `buildVirtualChart` method would be limited in scope to doing just calculations rather than render related logic (but it can do color scale calculation logic).